### ISSUE TITLE
Charset issue after whitenoise

### ISSUE THREAD

**Issue by jxrossel** [evansd/whitenoise#120]
Hi,

I'm trying to serve this file with whitenoise (from the visjs library):
https://cdnjs.cloudflare.com/ajax/libs/vis/4.18.1/vis.min.js

Using Django static server, I don't have any issue. On my prod website, with whitenoise (and without encoding / compression), the obtained file is slightly different (due to some special characters towards the end) and the execution fails. I have seen that we can set the charset in Whitenoise, but utf-8 is fine in general.

[vis.min_after_whitenoise.txt](https://github.com/evansd/whitenoise/files/764140/vis.min_after_whitenoise.txt)
[vis.min.txt](https://github.com/evansd/whitenoise/files/764141/vis.min.txt)

I use Django (1.10.5), Channels (1.0.3) and Whitenoise (3.3.0)

Any idea ?

**Comment by evansd** [comment]
Interesting, I've not come across this before. Can you reproduce the behaviour locally by running:
```
./manage.py runserver --nostatic`
```
I wonder if it's something to do with Channels. Which Python version are you using?

**Comment by jxrossel** [comment]
I'm using python 3.5. I'll check tomorrow what happens with --nostatic

**Comment by jxrossel** [comment]
Hi,

I've made some more testing to target the origin of this issue (dev / prod = settings choice):

- dev / runserver / no whitenoise: OK
- dev / runserver / no whitenoise / nostatic : NOK (as expected)
- dev / runserver / whitenoise / nostatic : OK
- prod (debug=False, Redis for cash and channels) / runserver / whitenoise after collectstatic : OK
- prod / daphne (with localhost) + workers (in circus) / whitenoise after collectstatic : OK ! (with code 200 for static files)
    - NB: admin website looks good as well
- prod / daphne (with non-localhost) + workers (in circus) / whitenoise after collectstatic : NOK (with 200 code for static files)
    - NB: admin website also damaged a bit

Basically, the problem arises when I pass from localhost to non-local host with the production settings. To go further, I disabled Channels and served my project with Waitress (pure wsgi). I got the same results: ok on localhost (127.0.0.1:8000), not on non-local host (0.0.0.0:8001).

For the record, I'm on Windows 7 (but it's not my fault).
